python/ray/serve/backend_state.py

```diff
@@ -36,63 +35,51 @@ class ReplicaState(Enum):
     STOPPED = 6
 
 
-class BackendReplica:
-    def __init__(self, controller_name: str, detached: bool,
+class ActorReplicaWrapper:
+    """Wraps a Ray actor for a backend replica.
+
+    This is primarily defined so that we can mock out actual Ray operations
+    for unit testing.
+
+    *All Ray API calls should be made here, not in BackendState.*
+    """
+
+    def __init__(self, actor_name: str, detached: bool, controller_name: str,
                  replica_tag: ReplicaTag, backend_tag: BackendTag):
-        self._actor_name = format_actor_name(replica_tag, controller_name)
+        self._actor_name = actor_name
         self._placement_group_name = self._actor_name + "_placement_group"
-        self._controller_name = controller_name
         self._detached = detached
+        self._controller_name = controller_name
         self._replica_tag = replica_tag
         self._backend_tag = backend_tag
-        self._actor_handle = None
-        self._placement_group = None
-        self._start_time = None
-        self._prev_slow_startup_warning_time = None
+
         self._startup_obj_ref = None
         self._drain_obj_ref = None
-        self._state = ReplicaState.SHOULD_START
+        self._stopped = False
         self._actor_resources = None
 
-    def __get_state__(self):
+        # Storing the handles is necessary to keep the actor and PG alive in
+        # the non-detached case.
+        self._actor_handle = None
+        self._placement_group = None
```

Comment:
Ahh, that is why these are still around. If the controller crashes in `detached=False`, what is the expected behaviour here?

Reply:
Good question. We should raise a good error message but right now I think it'll just complain with some very strange errors.

Reply:
and yeah, I originally removed them and everything broke...